### DISCUSSION TITLE
Parse array from REST data sourse and send array to REST data sourse

### DISCUSSION THREAD

**Discussion by Pharoenites** [Budibase/budibase#5507] [Help] [answered]
Hi all, help me with my project please.
I create button which have action "send POST request" with body from REST data source:
```
{
  "ICDCode": "{{ ICDCode }}"
}
```

My service send me response:
```
{
  "ICDCode": "code",
  "Parameters": [
    {
      "ID": ID,
      "ParameterName": "name",
      "ParameterType": "type",
      "URLForCalculation": "URL",
      "RangeOfValueType": "valueType",
      "SelectionType": "selectionType",
      "DimensionType": "dimensionType"
    },
    {
      "ID": ID,
      "ParameterName": "name",
      "ParameterType": "type",
      "URLForCalculation": "URL",
      "RangeOfValueType": "valueType",
      "SelectionType": "selectionType",
      "DimensionType": "dimensionType"
    },
... <many objects of array>
  ]
}
```

1. How can I parse this response on UI form? Desirable in table view.
2. When this data will be in the form of table, I want to add new attribute (Have Actual) and send this new array to next service in POST request.
Can I take data from form but not like `{{ New Form.Fields.ICDCode }}` but this way: `{{ New Form.<table with all fields>` to array for request body.

**Comment by melohagan** [comment]
Hi @Pharoenites 

1. You'll want to transform the response: https://docs.budibase.com/docs/transformers
    In this case I think it'll be straightforward, something like `return data.Parameters` assuming you want to have each parameter as a row in the table.
    You'll be able to see if the schema matches what you want: https://docs.budibase.com/docs/rest-queries#modifying-the-schema

2. Have you tried `{{ New Form.Fields }}` ?

**Comment by Pharoenites** [comment]
Thank you for answer
`return data.Parameters` returns only first object of array as if contain `return data.Parameters[0]` but not all array

How I can get all array and show it on UI?
If I type `{{ State.Res.Parameters  }}` or `{{ New Repeater.<my_request>.Parameters }}` or  `{{ New Cards block.<my_request>.Parameters }}` it returns `[object Object],[object Object],[object Object],[object Object],...`

If I use table block and set `<my_request>` as data it returns **1 row** with **2 columns**: `ICDCode (contains correct value of ICDCode)` and `Parameters (contains array "Parameters")` but I need table with **rows=countity objects of array** with **7 columns from object Parameter**

Also after this view I need add new column `Have Actual`, but it is another story

Help please :(

**Reply by melohagan** [reply]
Hey @Pharoenites When you say `return data.Parameters` only returns the first object, you are probably talking about the Schema, which would be correct as this is the Schema of each row.

**Comment by Pharoenites** [comment, marked as answer]
Thank you, I figured it out.
